subprojects/parseq-guava-interop/src/main/java/com/linkedin/parseq/guava/ListenableFutureUtil.java

```diff
@@ -40,7 +41,7 @@ public boolean cancel(Exception rootReason) {
           }
         };
 
-    final SettablePromise<T> promise = task.getSettableDelegate();
+    final SettablePromise<T> promise = task.getSettablePromise();
```

Comment:
Personally, I prefer the style we used in implementing Task.fromCompletableStage (https://github.com/linkedin/parseq/blob/master/subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java#L1335) by directly creating an instance of BaseTask by overriding its `run()` and `cancel()` methods. In `run()`, we can create a SettablePromise and addListener to the ListenerFuture, then we don't need to have this SettableTask at all and a private variable to hold that newly created promise.

Reply:
adopted

Reply:
adopted the suggestion